app/src/main/java/com/example/watpool/services/firebase_services/FirebaseTripsService.kt

```diff
@@ -0,0 +1,203 @@
+
+class FirebaseTripsService: TripsService {
+    private val database = Firebase.firestore
+    private val tripsRef: CollectionReference = database.collection("trips")
+    private val tripsConfirmationRef: CollectionReference = database.collection("trips_confirmation")
+
+    enum class DayOfTheWeek(val day: String) {
+        SUNDAY ("SUNDAY"), MONDAY ("MONDAY"), TUESDAY ("TUESDAY"),
+        WEDNESDAY ("WEDNESDAY"), THURSDAY ("THURSDAY"), FRIDAY ("FRIDAY"), SATURDAY ("SATURDAY")
+    }
+
+    // fetches the last trip and creates a clone
+//    override fun cloneTrip(tripId: String): Task<DocumentReference> {
+//        val oldTrip = tripsRef.whereEqualTo("id", tripId).get()
+//        return oldTrip
+//    }
+
+    // driver posting a new trip
+    @RequiresApi(Build.VERSION_CODES.O)
+    override fun createTrip(driverId: String, startingCoordinateId: String, endingCoordinateId: String, tripDate: LocalDate, maxPassengers: String, isRecurring: Boolean, recurringDayOfTheWeek: DayOfTheWeek, recurringEndDate: LocalDate): Task<DocumentReference> {
+        val id: String = UUID.randomUUID().toString()
+        val trip = hashMapOf(
+            "id" to id,
+            "starting_coordinate" to startingCoordinateId,
+            "ending_coordinate" to endingCoordinateId,
+            "max_passengers" to maxPassengers,
+            "is_recurring" to isRecurring,
+            "trip_date" to tripDate.toString(),
+            "driver_id" to driverId
+        )
+
+        if (isRecurring) {
+            trip["recurring_day"] = recurringDayOfTheWeek.day
+            trip["recurring_end_dat"] = recurringEndDate.toString()
+        }
+
+        return tripsRef.add(trip)
+    }
+
+    // rider requesting new ride
+    @RequiresApi(Build.VERSION_CODES.O)
+    override fun createTripPosting(userId: String, startingCoordinateId: String, endingCoordinateId: String, tripDate: LocalDate, isRecurring: Boolean, recurringDayOfTheWeek: DayOfTheWeek, recurringEndDate: LocalDate): Task<DocumentReference> {
+        val id: String = UUID.randomUUID().toString()
+        val trip = hashMapOf(
+            "id" to id,
+            "starting_coordinate" to startingCoordinateId,
+            "ending_coordinate" to endingCoordinateId,
+            "is_recurring" to isRecurring,
+            "trip_date" to tripDate.toString(),
+            "passengers" to listOf<String>(userId)
+        )
+
+        if (isRecurring) {
+            trip["recurring_day"] = recurringDayOfTheWeek.day
+            trip["recurring_end_date"] = recurringEndDate.toString()
+        }
+
+        return tripsRef.add(trip)
+    }
+
+
+    override fun createTripConfirmation(tripId: String, confirmationDate: LocalDate, riderIds: List<String>): Task<DocumentReference> {
+        val id: String = UUID.randomUUID().toString()
+        val confirmation = hashMapOf(
+            "id" to id,
+            "tripId" to tripId,
+            "confirmation_date" to confirmationDate.toString(),
```

Comment:
If we're going  to have a confirmation date, shouldn't we make the rider_ids field singular rather than being a list of users since users can confirm the trip at different times?